packages/graphql-fixtures/src/fill.ts

```diff
@@ -148,33 +149,56 @@ export function createFiller(
 
   const context = {schema, resolvers};
 
-  return function fill<Data, Variables, PartialData>(
-    _document: GraphQLOperation<Data, Variables, PartialData>,
-    data?: GraphQLFillerData<GraphQLOperation<Data, Variables, PartialData>>,
-  ): (request: GraphQLRequest<Data, Variables, PartialData>) => Data {
-    return (request: GraphQLRequest<Data, Variables, PartialData>) => {
-      const {operationName, query} = request;
-
-      const operation =
-        (operationName && documentToOperation.get(operationName)) ||
-        Object.values(compile(schema, normalizeDocument(query)).operations)[0];
-
-      if (operationName != null) {
-        documentToOperation.set(operationName, operation);
-      }
+  return {
+    fillFragment<Data, Variables, PartialData>(
+      type: DocumentNode<Data, Variables, PartialData>,
+      data?: GraphQLFillerData<GraphQLOperation<Data, Variables, PartialData>>,
+    ): Data {
+      const fragment = Object.values(
+        compile(schema, normalizeDocument(type)).fragments,
+      )[0];
 
       return fillObject(
-        operation.rootType,
-        operation.rootType,
-        // the root type is kind of weird, since there is no "field" that
-        // would be used in a resolver. For simplicity in the common case
-        // we just hack this type to make it conform.
-        [operation as any],
+        fragment.possibleTypes[0],
+        fragment.possibleTypes[0],
```

Comment:
To match the filling the way the rest of the autofilling stuff works, I think you'd need to randomly pick the concrete type you are filling for, rather than just taking the first one?